### PR TITLE
feat(mobile): add biometric plugin

### PR DESCRIPTION
**Pull request by lucasfernog-crabnebula**
This PR adds a new plugin that allows users to authenticate using the mobile device's biometry (or face ID).

### CODE REVIEW

File: plugins/biometric/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "tauri-plugin-biometric"
+version = "1.0.0"
```
Comment (amrbashir) [inline plugins/biometric/Cargo.toml:3]: is there a reason why this is not `2.0.0-alpha-0` or did you forget add a change file with `major` bump?

File: plugins/biometric/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+edition = { workspace = true }
+authors = { workspace = true }
+license = { workspace = true }
+links = "tauri-plugin-biometric"
+
+[package.metadata.docs.rs]
+features = [ "tauri/dox" ]
```
Comment (amrbashir) [inline plugins/biometric/Cargo.toml]: didn't we remove `tauri/dox` feature?

File: plugins/biometric/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        .plugin(tauri_plugin_biometric::init())
+        .run(tauri::generate_context!())
+        .expect("error while running tauri application");
+}
+```
+
+Afterwards all the plugin's APIs are available through the JavaScript guest bindings:
+
+```javascript
+
```
Comment (amrbashir) [inline plugins/biometric/README.md]: missing example

**Review by amrbashir — CHANGES_REQUESTED** [review]
Android code LGTM and tested as well, nice Work, I will leave iOS to you as usual.

File: plugins/biometric/guest-js/index.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+export async function checkStatus(): Promise<Status> {
+  return invoke("plugin:biometric|status");
+}
+
+export async function authenticate(
+  reason: string,
+  options?: AuthOptions,
+): Promise<void> {
```
Comment (amrbashir) [inline plugins/biometric/guest-js/index.ts:72]: Shouldn't this return whether it is successful or not?

File: plugins/biometric/Cargo.toml (hunk 1 above)
Comment (lucasfernog-crabnebula) [reply plugins/biometric/Cargo.toml:3]: i just forgot to add the change file, then when we release covector turns it into 2.0.0-alpha.0

File: plugins/biometric/guest-js/index.ts (hunk 4 above)
Comment (lucasfernog-crabnebula) [reply plugins/biometric/guest-js/index.ts:72]: it throws instead, added documentation for it

File: plugins/biometric/Cargo.toml (hunk 2 above)
Comment (lucasfernog-crabnebula) [reply plugins/biometric/Cargo.toml]: oops

**Review by amrbashir — DISMISSED** [review]

**Review by amrbashir — APPROVED** [review]

**Pull request merged**